### ISSUE TITLE
Ember 2.17 Breaking 

### ISSUE THREAD

**Issue by hhff** [rwjblue/ember-getowner-polyfill#28]
Hi Robert!

Thanks for all of the hard work here!

I just updated a few dependencies for an Ember App (including Ember 2.14 -> 2.17) I work on, but I've hit a snag that seems related to this library.  I believe it's the same issue reported here:

https://github.com/jamesarosen/ember-i18n/issues/466

Basically, with the removal of `_lookupFactory`, I'm now seeing the below in the console:
```
index.js:9 Uncaught TypeError: this[SAFE_LOOKUP_FACTORY_METHOD] is not a function
    at Class.factoryFor (index.js:9)
    at Class.superWrapper [as factoryFor] (ember-utils.js:437)
    at Class._initDefaults (i18n.js:82)
    at sendEvent (ember-metal.js:384)
    at new Class (core_object.js:119)
    at Function._ClassMixinProps.create (core_object.js:282)
    at FactoryManager.create (container.js:384)
    at Proxy.create (container.js:135)
    at instantiateFactory (container.js:224)
    at lookup (container.js:182)
```

Which I believe is triggered here:
https://github.com/rwjblue/ember-factory-for-polyfill/blob/master/vendor/ember-factory-for-polyfill/index.js#L9

However, when running `yarn why ember-factory-for-polyfill`, it's outputting:

```
dash [feature/369-refine-track-uploader●] % yarn why ember-factory-for-polyfill
yarn why v1.1.0
warning ../package.json: No license field
[1/4] 🤔  Why do we have the module "ember-factory-for-polyfill"...?
[2/4] 🚚  Initialising dependency graph...
[3/4] 🔍  Finding dependency...
[4/4] 🚡  Calculating file sizes...
info This module exists because "ember-concurrency#ember-getowner-polyfill" depends on it.
info Disk size without dependencies: "188kB"
info Disk size with unique dependencies: "236kB"
info Disk size with transitive dependencies: "308kB"
info Number of shared dependencies: 2
```

Do you happen to know why is this happening, and what I need to do to fix it?

Thankyou!

**Comment by rwjblue** [comment]
Hmm, seems like you may have an older version of this library being depended on.  The current release versions of both this addon and ember-factory-for-polyfill do absolutely nothing (no modification to the final assets at all) when consumed with more recent versions of Ember. The code you linked to should _never_ run on Ember 2.12+...

**Comment by hhff** [comment]
Thankyou Robert!  Yes - I see the code only importing that lib when the ember version is under 2.12 beta 1.

In fact, I have just fixed it.

In this project, I have a private addon that is shared between a couple apps (and also relies on libs that have `ember-getowner-polyfill`).  I just updated its version of `ember-source` to 2.12+ and everything is working as expected now.

I had neglected trying that as I assumed the Ember version that would be respected in the case of the ember-getowner-polyfill would be the host app's Ember version! 

Definitely a weird case, but hopefully this helps someone!
